### PR TITLE
perf(app): Use maps instead of in-RAM SQL DB (#260)

### PR DESCRIPTION
**Pull request by eaton-coreymutter**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-virtual-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [X] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [X] I have opened a PR for the related docs change (if not, why?)  edgex-docs/pull/677  

## Testing Instructions
I have built and ran on arm32, it's working in an existing setup with no visible changes, and ran unit tests on x86_64 with no problems.

## New Dependency Instructions (If applicable)
Actually removed dependencies, then did go mod tidy in merge commit

Closes: #260

### CODE REVIEW

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]
The RWMutex should be used for lock and unlock when accessing the map to avoid race condition

**Comment by cloudxxx8** [comment]
this document mentions about using ql, so it needs to be modified
https://docs.edgexfoundry.org/2.2/microservices/device/virtual/Ch-VirtualDevice/

**Comment by eaton-coreymutter** [comment]
Good catch, mutex added and tested.

**Review by FelixTing — APPROVED** [review]
LGTM

**Comment by eaton-coreymutter** [comment]
Added PR in edgex-docs, #677.

**Comment by eaton-coreymutter** [comment]
Don't think I can change the message on my latest commit to be semanticer, do I need to do another PR with a fresh branch?

**Comment by lenny-intel** [comment]
@eaton-coreymutter , Thank you for the contribution!!

A couple nits on the PR process. 

- Please amend your last commit to use conventional commit format (this is the cause for the failing verify check )
- Also please update the reference to the doc PR in the check list to be `edgex-docs/pull/677 ` so it becomes a live link.  `#pr-num` only works if the PR is in the same repo.

**Comment by eaton-coreymutter** [comment]
Thanks @lenny-intel, I hadn't realized I could update the message and force-push.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

File: internal/driver/db.go [hunk 1]
```diff
@@ -1,142 +1,117 @@
...
-	if db.connection == nil {
-		return fmt.Errorf("Lost DB connection, forgot to openDb()? ")
-	}
-	if tx, err := db.connection.Begin(); err != nil {
-		return err
-	} else {
-		db.transaction = tx
-		return nil
-	}
+func (db *db) openDb() error {
```
Comment (cloudxxx8) [inline internal/driver/db.go:42]: we can really remove this function and all the code calling this function

File: internal/driver/db.go [hunk 2]
```diff
@@ -1,142 +1,117 @@
...
 	DataType            string
 	Value               string
 }
 
+// Global: the data we are tasked with storing
+// Outer key: device name, inner key: resource name, value: resource info
+var resources map[string]map[string]data
+
+// Golang maps alone are not safe for concurrent access
+var resources_lock sync.RWMutex
```
Comment (cloudxxx8) [inline internal/driver/db.go:21]: The `resources` and `resources_lock` should be moved to the db struct
avoid to use the global variables

File: go.mod [hunk 3]
```diff
@@ -3,7 +3,6 @@ module github.com/edgexfoundry/device-virtual-go
 require (
 	github.com/edgexfoundry/device-sdk-go/v2 v2.2.0-dev.8
 	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.1.0
-	modernc.org/ql v1.0.0
```
Comment (cloudxxx8) [inline go.mod:6]: please help merge these two require section

File: go.mod [hunk 4]
```diff
@@ -48,24 +46,12 @@ require (
...
-	modernc.org/db v1.0.3 // indirect
-	modernc.org/file v1.0.3 // indirect
-	modernc.org/fileutil v1.0.0 // indirect
-	modernc.org/golex v1.0.1 // indirect
-	modernc.org/internal v1.0.2 // indirect
-	modernc.org/lldb v1.0.2 // indirect
-	modernc.org/mathutil v1.4.1 // indirect
-	modernc.org/sortutil v1.1.0 // indirect
-	modernc.org/strutil v1.1.1 // indirect
-	modernc.org/zappy v1.0.3 // indirect
```
Comment (cloudxxx8) [inline go.mod:68]: please help remove all the libraries you deleted from https://github.com/edgexfoundry/device-virtual-go/blob/d36d4eb5d49c50cebb55fcc728bec397aa4f388e/Attribution.txt

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: internal/driver/db.go (hunk 1 above)
Comment (eaton-coreymutter) [reply internal/driver/db.go]: Removed.

File: internal/driver/db.go (hunk 2 above)
Comment (eaton-coreymutter) [reply internal/driver/db.go]: Brought them into the struct and it still works.

File: go.mod (hunk 4 above)
Comment (eaton-coreymutter) [reply go.mod:68]: Removed - I checked against what was present after "go mod vendor".

Comment (cloudxxx8) [reply go.mod:68]: sorry, that how go 1.17 works, and I didn't notice that

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**